### PR TITLE
Redirect donor to front page on file input error

### LINKED ISSUE #116: submitting a file that isn't an image break the app, but doesn't tell you why
we should pop up an alert instead

#### From the Heroku logs

2015-03-04T01:52:37.168690+00:00 app[web.1]:
2015-03-04T01:52:37.168697+00:00 app[web.1]:   app/forms/donation_form.rb:40:in `rescue in create_objects'
2015-03-04T01:52:37.168703+00:00 app[web.1]:   app/controllers/donations_controller.rb:13:in `block in create'
2015-03-04T01:52:37.168695+00:00 app[web.1]: NameError (undefined local variable or method `logger' for #<DonationForm:0x007f543554c820>):
2015-03-04T01:52:37.168705+00:00 app[web.1]:   app/controllers/donations_controller.rb:12:in `create'
2015-03-04T01:52:37.168700+00:00 app[web.1]:   app/forms/donation_form.rb:36:in `create_objects'

### PR DESCRIPTION
**Pull request by nikitr**
Fix #116

### CODE REVIEW

File: app/forms/donation_form.rb [hunk 1]
```diff
@@ -37,7 +37,7 @@ def create_objects
       donation.save!
     end
     rescue ActiveRecord::RecordInvalid => err
-      logger.error(err.to_s)
+      # logger.error(err.to_s)
```
Comment (nikitr) [inline app/forms/donation_form.rb]: @jjwon this was throwing a bug and is unnecessary, what should I do about it?

Comment (jjwon) [inline app/forms/donation_form.rb]: I think removing the entire rescue block is okay.

File: app/controllers/recipients_controller.rb [hunk 2]
```diff
@@ -5,6 +5,7 @@ def new
 
   # POST /receive
   def create
+    authorize! :create, @interest, message: "Not authorized to receive"
```
Comment (jjwon) [inline app/controllers/recipients_controller.rb]: what does this have to do with the problem?

File: app/controllers/donations_controller.rb [hunk 3]
```diff
@@ -7,6 +7,7 @@ def new
 
   # POST /donate
   def create
+    authorize! :create, Donation, message: "Not authorized to donate"
```
Comment (jjwon) [inline app/controllers/donations_controller.rb]: ehh not sure why this is here

File: app/controllers/donations_controller.rb [hunk 4]
```diff
@@ -18,7 +19,7 @@ def create
         UserMailer.coordinator_email(@coordinators, @donation).deliver
         format.html { redirect_to root_path, notice: "Donation was successfully created." }
       else
-        format.html { render donations_new_path }
+        format.html { redirect_to root_path, notice: "Donation failed because of unsupported file type." }
```
Comment (jjwon) [inline app/controllers/donations_controller.rb]: seems to work!!!

however change it from a notice to an alert, so the notice color is red instead.

**Comment by nikitr** [comment]
Changed notice to an alert and removed unnecessary parts
